src/tmpfiles/tmpfiles.c

```diff
@@ -2021,8 +2194,14 @@ static int create_item(Item *i) {
 
         case TRUNCATE_FILE:
         case CREATE_FILE:
-                RUN_WITH_UMASK(0000)
-                        (void) mkdir_parents_label(i->path, 0755);
+                if (i->try_replace) {
+                        r = mkdir_parents_rm_if_wrong_type(S_IFREG, i->path);
+                        if (r < 0 && r != -ENOENT)
+                                return r;
+                } else {
+                        RUN_WITH_UMASK(0000)
+                                (void) mkdir_parents_label(i->path, 0755);
+                }
```

Comment:
please move the mkdir_parents_label() call into mkdir_parents_rm_if_wrong_type() too, after all you have the same combination of if branches here 7 times!

Reply:
I put it in an intermediate helper function to avoid making mkdir_parents_rm_if_wrong_type even more complex.